### PR TITLE
fix(lib-injection): build wheels for dependencies

### PR DESCRIPTION
**Pull request by majorgreys**
This fixes the lib-injection build which broke with #6651.

Here is the results locally:

```
$ docker run --rm -it (docker build --build-arg DDTRACE_PYTHON_VERSION=2.0.2 -q . ) ls -1 ddtrace_pkgs
site-packages-ddtrace-py3.10-manylinux2014
site-packages-ddtrace-py3.10-musllinux_1_1
site-packages-ddtrace-py3.11-manylinux2014
site-packages-ddtrace-py3.11-musllinux_1_1
site-packages-ddtrace-py3.12-manylinux2014
site-packages-ddtrace-py3.12-musllinux_1_1
site-packages-ddtrace-py3.7-manylinux2014
site-packages-ddtrace-py3.7-musllinux_1_1
site-packages-ddtrace-py3.8-manylinux2014
site-packages-ddtrace-py3.8-musllinux_1_1
site-packages-ddtrace-py3.9-manylinux2014
site-packages-ddtrace-py3.9-musllinux_1_1
```

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [ ] Title is accurate.
- [ ] No unnecessary changes are introduced.
- [ ] Description motivates each change.
- [ ] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [ ] Testing strategy adequately addresses listed risk(s).
- [ ] Change is maintainable (easy to change, telemetry, documentation).
- [ ] Release note makes sense to a user of the library.
- [ ] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [ ] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)
- [ ] If this PR touches code that signs or publishes builds or packages, or handles credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [ ] This PR doesn't touch any of that.

### CODE REVIEW

File: lib-injection/Dockerfile [hunk 1]
```diff
@@ -9,22 +25,52 @@ RUN mkdir -p pkgs
...
+RUN python3 -m pip install packaging==23.0
+RUN mkdir -p pkgs
+ADD ./dl_wheels.py .
+# Note that we only get Python >= 3.7. This is to keep the size of the image
+# as small as possible.
+RUN python3 dl_wheels.py \
         --python-version=3.10 \
-        --python-version=3.9 \
-        --python-version=3.8 \
-        --python-version=3.7 \
```
Comment (Kyle-Verhoog) [inline lib-injection/Dockerfile:17]: It looks like we're not supporting these now? We'll need to add these back in along with the different arch's

File: lib-injection/Dockerfile [hunk 2]
```diff
@@ -9,22 +25,52 @@ RUN mkdir -p pkgs
 ADD ./dl_wheels.py .
 # Note that we only get Python >= 3.7. This is to keep the size of the image
 # as small as possible.
+RUN python3 dl_wheels.py \
```
Comment (Kyle-Verhoog) [inline lib-injection/Dockerfile:28]: this stage can be merged with the previous by using multiple `--python-version`s

File: lib-injection/Dockerfile [hunk 3]
```diff
@@ -9,22 +25,52 @@ RUN mkdir -p pkgs
...
+# Note that we only get Python >= 3.7. This is to keep the size of the image
+# as small as possible.
 RUN python3 dl_wheels.py \
         --python-version=3.11 \
+        --platform="musllinux_1_1" \
+        --ddtrace-version=${DDTRACE_PYTHON_VERSION} \
+        --output-dir /build/pkgs \
+        --verbose
+
+FROM python:3.10-alpine
```
Comment (Kyle-Verhoog) [inline lib-injection/Dockerfile:52]: can we merge this stage with the previous? Any reason to have to do it per python version?

**Review by Kyle-Verhoog — COMMENTED** [review]
@majorgreys would the dd-trace-py test image container contain all the dependencies required? Not entirely sure it would work because of the alpine case still.

That would mean just a change of the base image.

I'm ok with the fix since I can't think of anything better, if the stages can be merged then we only need 1 other stage for alpine/musl

File: lib-injection/Dockerfile (hunk 3 above)
Comment (majorgreys) [reply lib-injection/Dockerfile:52]: my conclusion was that we'd need multiple stages in order to build the wheels with `pip wheels` for each architecture and platform.

**Review by Kyle-Verhoog — CHANGES_REQUESTED** [review]
https://github.com/DataDog/dd-trace-py/blob/fa3a486f09a168bb5cc94bbf3bc5494116aad796/.gitlab/build-deb-rpm.sh needs to be updated as well if we're going to do away with `dl_wheels.py`. As it stands, the deb/rpm packages will be broken.

Also can we use the dd-trace-py `testrunner` image so that we can build all the manylinux wheels together in one stage? This might help reduce the duplication at least for manylinux. Ideally we could find a musl-based image that provides all the required Python versions as well.

File: .gitlab/build-deb-rpm.sh [hunk 4]
```diff
@@ -10,26 +10,14 @@ source common_build_functions.sh
...
 
-../lib-injection/dl_wheels.py \
-    --python-version=3.11 \
-    --python-version=3.10 \
-    --python-version=3.9 \
-    --python-version=3.8 \
-    --python-version=3.7 \
-    --ddtrace-version=$PYTHON_PACKAGE_VERSION  \
-    --arch x86_64 \
-    --arch aarch64 \
```
Comment (majorgreys) [inline .gitlab/build-deb-rpm.sh:28]: @Kyle-Verhoog Can you confirm we do not need to worry about aarch64 here?

File: .gitlab/build-deb-rpm.sh [hunk 5]
```diff
@@ -10,26 +10,14 @@ source common_build_functions.sh
...
-    --platform musllinux_1_1 \
```
Comment (majorgreys) [inline .gitlab/build-deb-rpm.sh:29]: @Kyle-Verhoog Do we need to build these for the deb/rpm packages?

File: .gitlab/build-deb-rpm.sh (hunk 4 above)
Comment (Kyle-Verhoog) [reply .gitlab/build-deb-rpm.sh:28]: We need to support ARM64

**Comment by emmettbutler** [comment]
@majorgreys how can I help with this?

**Pull request closed without merging**